src/controllers/sendResetToken.js

```diff
@@ -0,0 +1,114 @@
+import { randomBytes } from 'crypto';
+import models from '../db/models';
+import sendgrid from '../middlewares/helper';
+
+const { User } = models;
+
+/**
+  * @description class representing reset user password
+  *
+  * @class SendResettoken
+  */
+class SendResetToken {
+  /**
+    * @description - This method is responsible for creating creating password reset token
+    *
+    * @static
+    * @param {object} request - Request sent to the router
+    * @param {object} response - Response sent from the controller
+    *
+    * @returns {object} - object representing response message
+    *
+    */
+  static async resetToken(request, response) {
+    try {
+      const { email } = request.body;
+      if (email.trim().length < 1) {
+        return response.status(400).json({
+          status: 'Fail',
+          message: 'The email field cannot be empty'
+        });
+      }
+      const userData = await User.findOne({
+        where: { email }
+      });
+
+      if (!userData) {
+        return response.status(400).json({
+          status: 'Fail',
+          message: 'An email has been sent to your mail',
+        });
+      }
+      const userToken = randomBytes(40).toString('hex');
+      const tokenExpiresIn = Date.now() + 3600000;
+
+      const { username, password } = userData.dataValues;
+      const userUpdatedData = await userData.update({
+        token: userToken,
+        username,
+        email,
+        password,
+        passwordExpiresIn: tokenExpiresIn || userData.passwordExpiresIn,
+      });
+      const { HOST_URL } = process.env;
+      await sendgrid(userData.email, 'noreply@authors.com', HOST_URL, userToken);
+
+      response.status(200).json({
+        status: 'Success',
+        message: 'Password reset link has been sent to your mail',
+        token: userUpdatedData.token,
+      });
+    } catch (error) {
+      response.status(500).json({
+        status: 'Error',
+        error: error.message
+      });
+    }
+  }
+
+  /**
+    * @description - This method is responsible for updating user password
+    *
+    * @static
+    * @param {object} request - Request sent to the router
+    * @param {object} response - Response sent from the controller
+    *
+    * @returns {object} - object representing response message
+    *
+    */
+  static async updateUserPassword(request, response) {
+    const { password, confirmNewPassword } = request.body;
+    const { token } = request.query;
+    try {
+      const userToUpdate = await User.findOne({
+        where: { token }
+      });
+      if (parseInt(userToUpdate.passwordExpiresIn, 10) < parseInt(Date.now(), 10)) {
```

Comment:
@okoroemeka I see that you're validating a time period of expiration here can this also be abstracted as a middleware?

Reply:
It has been done using JWT.